minio/minio
# Introduce new config `subnet api_key`

## Description

The earlier approach of using a license token for communicating with
SUBNET is being replaced with a simpler mechanism of api keys. Unlike
the license which is a JWT token, these api keys will be simple UUID
tokens, and don't have any embedded information in them. SUBNET would
generate the api key on cluster registration, and then it would be saved
in this config, to be used for subsequent communication with SUBNET.

## Motivation and Context

Simplify communication between minio/mc and SUBNET

## How to test this PR?

- Build minio using this PR
- Create a cluster using this minio binary
- Build mc using [PR #3876](https://github.com/minio/mc/pull/3876)
- Start SUBNET server locally
- Register the cluster with SUBNET by running `./mc admin subnet register {alias} --dev`
  - This will ask for SUBNET credentials. Provide the same (including email OTP)
- Registration should succeed with message `mc: Cluster {alias} successfully registered on SUBNET.`
- Check that API token has got saved in minio config using `./mc admin config get {alias} subnet`
- Running the registration command again should work without SUBNET credentials now

## Types of changes

## Checklist:
